recipes/keychain/all/conanfile.py

```diff
@@ -1,26 +1,27 @@
-from conans import ConanFile, CMake, tools
+from conan import ConanFile
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+from conan.tools.files import collect_libs, copy, get
+from conan.tools.gnu import PkgConfigDeps
+
 import os
 
+
 class KeychainConan(ConanFile):
     name = "keychain"
     homepage = "https://github.com/hrantzsch/keychain"
     description = "A cross-platform wrapper for the operating system's credential storage"
-    topics = ("conan", "keychain", "security", "credentials", "password", "cpp11")
+    topics = ("keychain", "security", "credentials", "password", "cpp11")
     url = "https://github.com/conan-io/conan-center-index"
     license = "MIT"
     settings = "os", "arch", "compiler", "build_type"
     exports_sources = ["CMakeLists.txt"]
-    generators = "cmake", "pkg_config"
     options = {'fPIC': [False, True]}
```

Comment:
The missing binaries are an interesting consequence of `dlib` having a `shared=True` default while all others have `shared=False` by default - for Conan 2.0, the CI pipeline no longer builds those configurations and will enforce either all shared or all static.

This would be fixed if this library had the `shared` option - does this library support it? is there a reason why it shouldn't be an option?

Reply:
There is not! I added the option to the recipe.

This revealed that the test package has to be adjusted to conan2 as well. I did that but without the old `conan_basic_setup()` magic I cannot get it to link to the dependencies anymore. I'm very much stuck and would greatly appreciate a hint.